### PR TITLE
Add region filter for image list

### PR DESCRIPTION
**Pull request by JargoonPard**
cleaned up some ignored tests and header copyright.
Added a custom filter for filtering Azure marketplace images and user custom images by region

### CODE REVIEW

File: app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/image.regional.filter.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.azure.serverGroup.configure.basicSettings.image.filter', [
+  require('../../../../../core/utils/lodash.js'),
+]).filter('regional', function(_) {
+  return function(input, selectedRegion) {
+    var result = _.filter(input, function(image) {
+      return image.region === selectedRegion || image.region == null;
```
Comment (anotherchrisberry) [inline app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/image.regional.filter.js]: I think the linter will complain about the `==` check not being `===`

Comment (anotherchrisberry) [inline app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/image.regional.filter.js]: I think the linter will complain about the `==` check not being `===`

File: app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/basicSettings.html [hunk 2]
```diff
@@ -51,8 +51,8 @@
         <div class="col-md-9">
           <ui-select class="form-control input-sm" required ng-model="command.imageName" on-select="basicSettingsCtrl.imageChanged($item)">
             <ui-select-match placeholder="Pick an image">{{$select.selected.imageName}}</ui-select-match>
-            <ui-select-choices repeat="image.imageName as image in command.images | anyFieldFilter: {imageName: $select.search, ami: $select.search } | orderBy: '-imageName'">
-              <span ng-bind-html="image.imageName | highlight: $select.search"></span>
+            <ui-select-choices repeat="image.imageName as image in command.images | regional:command.region ">
```
Comment (anotherchrisberry) [inline app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/basicSettings.html]: Does this still filter the images based on the user typing into the search field?

Comment (JargoonPard) [inline app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/basicSettings.html]: No, we haven't enabled full marketplace search yet and are currently restricting the list to a small subset provided in clouddriver config so I have removed the search feature until we add in full marketplace images.

Comment (anotherchrisberry) [inline app/scripts/modules/azure/serverGroup/configure/wizard/basicSettings/basicSettings.html]: Sorry, I meant the filtering that happens when the user types into the search field, e.g.
screen shot 2016-03-08 at 9 00 08 am
I think you just need another filter after the `regional` filter:
```html
<ui-select-choices repeat="image.imageName as image in command.images | regional:command.region | filter:$select.search">
```
Alternatively, if it's a small list, you can omit the search functionality entirely from the select box by setting `search-enabled: false` on the `<ui-select>` tag.

**Comment by JargoonPard** [comment]
Updated to include the search filter.  thanks @anotherchrisberry

**Pull request merged**